<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CMainWindow</class>
 <widget class="QDialog" name="CMainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>429</width>
    <height>308</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>CMainWindow</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="gbParameters">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Experiment parameters</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="lbMinN">
        <property name="text">
         <string>Min N</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QSpinBox" name="sbMinN">
        <property name="minimum">
         <number>1024</number>
        </property>
        <property name="maximum">
         <number>999999999</number>
        </property>
        <property name="singleStep">
         <number>1024</number>
        </property>
        <property name="value">
         <number>1024</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="lbMaxN">
        <property name="text">
         <string>Max N</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="sbMaxN">
        <property name="minimum">
         <number>1024</number>
        </property>
        <property name="maximum">
         <number>999999999</number>
        </property>
        <property name="singleStep">
         <number>1024</number>
        </property>
        <property name="value">
         <number>1048576</number>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="lbMinThreadsPerBlock">
        <property name="text">
         <string>Min threads per block</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="sbMinThreadsPerBlock">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1024</number>
        </property>
        <property name="value">
         <number>1</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="lbMaxThreadsPerBlock">
        <property name="text">
         <string>Max threads per block</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="sbMaxThreadsPerBlock">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1024</number>
        </property>
        <property name="singleStep">
         <number>1</number>
        </property>
        <property name="value">
         <number>1024</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="lbOriginalsFilePath">
        <property name="text">
         <string>Originals file path:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <property name="spacing">
          <number>0</number>
         </property>
         <property name="margin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="leOriginalsFilePath">
           <property name="text">
            <string>elapsedTime.m</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="btnChooseOriginalsFile">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>...</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="btnLoadOriginals">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Load into UI</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="lbUseOriginals">
        <property name="text">
         <string>Use originals?</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="cbUseOriginals">
        <property name="text">
         <string>Use originals</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="btnExecExperiment">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Execute experiment</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QFrame" name="frmProgress">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="lbProgress">
        <property name="text">
         <string>Progress:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QProgressBar" name="pbProgress">
        <property name="value">
         <number>0</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnCancelExperiment">
        <property name="text">
         <string>Cancel experiment</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QLabel" name="lbStatus">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>Ready...</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <tabstops>
  <tabstop>btnExecExperiment</tabstop>
  <tabstop>btnCancelExperiment</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>btnExecExperiment</sender>
   <signal>clicked()</signal>
   <receiver>CMainWindow</receiver>
   <slot>executeClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>44</x>
     <y>219</y>
    </hint>
    <hint type="destinationlabel">
     <x>3</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnCancelExperiment</sender>
   <signal>clicked()</signal>
   <receiver>CMainWindow</receiver>
   <slot>cancelClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>360</x>
     <y>258</y>
    </hint>
    <hint type="destinationlabel">
     <x>8</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnChooseOriginalsFile</sender>
   <signal>clicked()</signal>
   <receiver>CMainWindow</receiver>
   <slot>chooseOriginalsFileClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>357</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnLoadOriginals</sender>
   <signal>clicked()</signal>
   <receiver>CMainWindow</receiver>
   <slot>loadOriginals()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>373</x>
     <y>357</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbUseOriginals</sender>
   <signal>toggled(bool)</signal>
   <receiver>CMainWindow</receiver>
   <slot>useOriginalsChanged(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>270</x>
     <y>332</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>executeClick()</slot>
  <slot>cancelClick()</slot>
  <slot>chooseOriginalsFileClick()</slot>
  <slot>loadOriginals()</slot>
  <slot>useOriginalsChanged(bool)</slot>
  <slot>experimentParameterChanged()</slot>
 </slots>
</ui>
